rfcs/20221116-enable-stochastic-rounding.md

```diff
@@ -0,0 +1,329 @@
+calls. Also, doing it this way, we can potentially avoid OOM issues introduced
+by storing and moving large random tensors by optimizing the backend
+implementations. Otherwise, the OOM issue could be inevitable when users need to
+generate random tensors and put them into the API.
+
+However, as more use cases emerge, there might be occasions where custom random
+number generators other than the default need to be provided. Or there might be
+experimental random number generation algorithms to be passed in. Under such
+circumstances, encapsulation might complicate future developments. However, up
+till now we haven’t received any user requests regarding random number
+generation algorithms or custom generators.
+
+### Alternatives Considered
+
+Similar to the proposal, but this option doesn't encapsulate random number generation
+inside of the API call and asks users to provide random tensors.
+
+```
+tf.stochastic_cast(input, randoms, dtype)
+```
+                                     
+<table>
+ <tr><td><strong>Args</strong></td><td><strong>Definition</strong></td></tr>
+ <tr><td>input</td><td>The original tensor to be casted</td></tr>
+ <tr><td>dtype</td><td>Desired type after casting</td></tr>
+ <tr><td>randoms</td><td>Random numbers for determining the rounding direction. If the random number is less than the fractional part, the  result will be rounded up. This is required to be unsigned integers whose bit width is the same as the operand.</td></tr>
+</table>
+
+
+Returns |                                                                 |
+------- | ---------------------------------------------------------------
+Tensor  | A tensor whose values are casted, with the same shape as input.
+
+This API asks users to provide random numbers so that users can control the
+rounding results as long as they can reproduce the sequence of random numbers,
+which can be achieved via tf.random.generator API. Depending on the use case,
+users can also achieve round-to-zero or round-away-from-zero by setting the
+random number to 0 or 0xFFFF (depending on the operand bit width).
+
+With rounding and downcasting encapsulated in one API, the op can be emulated in
+the backend with higher efficiency.
+
+There is a disadvantage of this implementation that users need to materialize
+large tensors before calling this API. Once they are
+materialized, calling this API will require storing and passing the large tensor
+around, causing OOM issues. When facing this OOM issue, users will need to potentially break down
+the tensor in order to bypass the memory limit.
+
+Another way that sticks to the current rounding API design:
+
+```
+tf.stochastic_round(input, to_precision, seed=None, alg=’Philox’)
+```
+
+<table>
+ <tr><td><strong>Args</strong></td><td><strong>Definition</strong></td></tr>
+ <tr><td>to_precision</td><td>Desired precision after rounding</td></tr>
+ <tr><td>seed</td><td>Required seed for the RNG.</td></tr>
```

Comment:
it has a default value 'None' so what happens if users do not set it? what is the guideline on how to choose a seed